sw/device/sca/lib/aes.h

```diff
@@ -0,0 +1,149 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+#ifndef OPENTITAN_SW_DEVICE_SCA_LIB_AES_H_
+#define OPENTITAN_SW_DEVICE_SCA_LIB_AES_H_
+
+/**
+ * NOTE: The only intended use of this code is to serve as a PRNG for generating
+ * input data for SCA experiments and penetration testing.
+ * The library is not hardened against any type of attacks, and it should not be
+ * used for any purpose other than stated.
+ *
+ * During the SCA experiments, encryptions are verified on the host side by
+ * running the same encryption using PyCryptodome package and comparing the
+ * result.
+ *
+ * Implementation of round-functions is based on a transposed-state technique
+ * for 32-bit architecture presented in:
+ *
+ * [1] Bertoni et. al., Efficient Software Implementation of AES on 32-Bit
+ *     Platforms, CHES 2002.
+ *
+ *     https://link.springer.com/content/pdf/10.1007/3-540-36400-5_13.pdf
+ *
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/memory.h"
+
+/**
+ * Encrypt one data block (16 Bytes) in ECB mode.
+ *
+ * @param  plain_text  Input block to enrypt
+ * @param  round_keys  All round keys (pre-computed)
+ * @param  cipher_text Encrypted output block
+ */
+void aes_sw_encrypt_block(const uint8_t *plain_text, const uint32_t *round_keys,
+                          uint8_t *cipher_text);
+
+/**
+ * Generate all round keys for AES-128 encryption.
+ * Store keys in a transposed-state form.
+ *
+ * @param  round_keys Round keys for all rounds
+ * @param  key        Encryption key
+ */
+void aes_key_schedule(uint32_t *round_keys, const uint8_t *key);
+
+static const uint8_t sbox[256] = {
```

Comment:
Const naming notation.
```suggestion
static const uint8_t kSbox[256] = {
```